mypy/nodes.py

```diff
@@ -398,10 +398,10 @@ def __init__(self) -> None:
         self.unanalyzed_type = None  # type: Optional[mypy.types.Type]
         # If method, reference to TypeInfo
         # TODO: Type should be Optional[TypeInfo]
-        self.info = cast(TypeInfo, None)
-        self.is_property = False
-        self.is_class = False
-        self.is_static = False
+        self.info = FUNC_NO_INFO  # type: TypeInfo
+        self.is_property = False  # type: bool
+        self.is_class = False  # type: bool
+        self.is_static = False  # type: bool
```

Comment:
Why do you need these three `bool`s here?

Reply:
Because referring to `FUNC_NO_INFO`, which is defined later in the file, causes `__init__` to get second-pass typechecked, which causes these to not get inferred right or something.
Annotating the FakeInfos fixes this and the weird `subtype.py` assert also so I'm doing that instead.